### PR TITLE
[agent-smith] Allow commands to skip infringements

### LINKED ISSUE #5280: Running brew update causes environment to stop
### Bug description

If I try to run `brew update` in an environment, it first runs for a while and then the environment stops.

### Steps to reproduce

1. Start a default Rails environment from the Gitpod examples
2. Run `brew update`

### Expected behavior

_No response_

### Example repository

https://github.com/gitpod-io/template-ruby-on-rails-postgres

### Anything else?

I wanted to update Homebrew because when I was trying to install the Stripe CLI with brew (`brew install stripe/stripe-cli/stripe`) it always returned that there's invalid syntax in the tap. Might this be related?

### PR DESCRIPTION
**Pull request by aledbf**
## Related Issue(s)

Fixes #5280

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **aledbf** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ee/agent-smith/pkg/agent/agent.go [hunk 1]
```diff
@@ -696,6 +717,15 @@ func (agent *Smith) handleExecveEvent(execve Execve) func() (*InfringingWorkspac
 			return nil, nil
 		}
 
+		if agent.allowList != nil {
```
Comment (princerachit) [inline components/ee/agent-smith/pkg/agent/agent.go:720]: Can you add tests for this please?

**Review by princerachit — CHANGES_REQUESTED** [review]

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **aledbf** after the PR has been reviewed.

Associated issue: *#5280*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by princerachit — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 3565cb27f895b33ca1569fb4091f5abb3a719c26</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6025#pullrequestreview-771090818" title="Approved">princerachit</a>*

Associated issue: *#5280*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)~~ [princerachit]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**